crates/context/config/src/types.rs

```diff
@@ -233,13 +290,44 @@ impl<T: serde::Serialize> Signed<T> {
         })
     }
 }
+
+pub trait IntoResult<T> {
+    type Error;
+
+    fn into_result(self) -> Result<T, Self::Error>;
+}
+
+impl<T> IntoResult<T> for T {
+    type Error = std::convert::Infallible;
+
+    fn into_result(self) -> Result<T, Self::Error> {
+        Ok(self)
+    }
+}
+
+impl<T, E> IntoResult<T> for Result<T, E> {
+    type Error = E;
+
+    fn into_result(self) -> Result<T, Self::Error> {
+        self
+    }
+}
```

Comment:
This seems a bit... strange. And without documentation, I can't work out what the motivation is. All it does is wrap something in a `Result`... Hmmmm.

Reply:
this is used in `Signed`, as a helper for the signing & verification processes, permitting a function to fail safely if it can as opposed to panicking, for example, while all byte slices are valid scalars in curve25519 to make up a `SigningKey`, not all of them are valid points in the curve to make up a `VerifyingKey`, but if you already had a `VerifyingKey` you won't have any errors in the key derivation process.

This is an accommodation for flexibility, if you have a result, return it, if not, return what you have and not have to worry about `rustc` asking for an `Err` type since we were too restrictive on the type system.